src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php

```diff
@@ -49,8 +50,12 @@ public function normalize($object, $format = null, array $context = [])
 
         $data = $this->normalizer->normalize($object, $format, $context);
 
+        if(gettype($data) !== 'array'){
+            return $data;
+        }
+
         try {
-            $data['price'] = $this->priceCalculator->calculate($object, ['channel' => $this->channelContext->getChannel()]);
+            $data['price'] = PriceTransformer::centimesToUnitsForNullableFields($this->priceCalculator->calculate($object, ['channel' => $this->channelContext->getChannel()]));
```

Comment:
Please try to fit the entire call code into one screen. By the way, read the PSR standards